main.py

```diff
@@ -71,18 +76,36 @@ def main(argv: Sequence[str] | None = None) -> int:
 
     os.chdir(args.pwd)
 
-    # generic/standalone tools
+    # generic/standalone tools that do not care about devenv configuration
     if args.command == "pin-gha":
         return pin_gha.main(remainder)
-    # Future home of bootstrap-sentry, and then bootstrap-*.
 
-    # the remaining tools are repo-specific
+    config = configparser.ConfigParser()
+    config_path = f"{config_root}/config.ini"
+
+    if not os.path.exists(config_path):
+        coderoot = os.path.expanduser(
+            input("please enter the root directory where you want to work in [~/dev]: ")
+        ) or os.path.expanduser("~/dev")
+        os.makedirs(coderoot, exist_ok=True)
+
+        config["devenv"] = {"coderoot": coderoot}
+
+        os.makedirs(config_root, exist_ok=True)
+        with open(config_path, "w") as f:
+            config.write(f)
+        print(f"If you made a mistake, you can edit {config_path}.")
 
-    # TODO: read a well-known json for preferences
-    coderoot = "dev"
+    config.read(config_path)
+    coderoot = config["devenv"]["coderoot"]
```

Comment:
We might want to check that the coderoot actually exists

Reply:
It doesn't have to exist for the bootstrap step (as bootstrap will be creating the coderoot if it doesn't exist)
and the subsequent `args.pwd.startswith(coderoot)` will fail if coderoot doesn't exist